### PR TITLE
[postgres-cluster] New chart to create Postgres clusters with local storage

### PR DESCRIPTION
**Pull request by angelnu**
**Description of the change**

This chart is a wrapper for the [Zalando postgres operator](https://github.com/zalando/postgres-operator) to create
a high available Postgres cluster using nodes local storage.

**Benefits**

Features added by this wrapper:
- creates an storage class using local storage in the nodes specified in `persistentVolumes.replicaNodes`. These replicas survive
  a tear-down and tear-up of the cluster.
- (optional) does sql dumps to an existing PVC with K8S cronjobs. The Operator only supports backups to cloud, not NFS PVC for example.
- Define the password for the DB so that it can be used accross tear-down and tear-up of the cluster.

**Possible drawbacks**

Another chart to maintain - I can offer the maintenance.

**Applicable issues**

- fixes #

**Additional information**

Since it depends on the operator being loaded previously and also knowing the topology of the cluster I have disabled the install test and only do lint.

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [x] (optional) Chart is using our [common](https://github.com/k8s-at-home/charts/tree/master/charts/common) library as a dependency.
- [x] (optional) Variables are documented in the README.md

### CODE REVIEW

File: charts/stable/postgres-cluster/templates/storageClass.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: {{ include "postgres-cluster.storageClass" . }}
+  labels:
+    {{- include "common.labels" . | nindent 4 }}
+  annotations:
+provisioner: kubernetes.io/no-provisioner
+reclaimPolicy: Delete
+volumeBindingMode: WaitForFirstConsumer
```
Comment (onedr0p) [inline charts/stable/postgres-cluster/templates/storageClass.yaml:10]: New at the end of the file here :)

**Review by onedr0p — CHANGES_REQUESTED** [review]
This looks like an interesting chart! Just one small change and I'll trust you to help maintain it :D

**Comment by onedr0p** [comment]
Do you think this should be called something like `zalando-postgres-cluster`?

**Comment by angelnu** [comment]
Yes, I can rename it - will do now together with the missing end of file new line.

Sure I will maintain it - I have 6 postgres clusters myself using this! I find easier to deal with DB issues and going back when the DB is used only by a single chart.

@onedr0p - if we consider this chart as a dependency for other charts in this repo - do we still it put it here or under the library repository?

**Comment by onedr0p** [comment]
I think leaving it here is fine, we can always change it in the future if we feel differently 🙂

**Review by onedr0p — APPROVED** [review]

**Pull request merged**